creator-node/src/apiSigning.js

```diff
@@ -23,6 +23,33 @@ const generateTimestampAndSignature = (data, privateKey) => {
   return { timestamp, signature: signedResponse.signature }
 }
 
+// Keeps track of cached signatures for `generateTimestampAndSignatureIfNecessary`
+const cachedSignatures = {}
+
+/**
+ * Generates a signature for `data` if only the previous signature
+ * generated is invalid (expired). Otherwise returns an existing signature.
+ * @param {string} data only string data supported
+ * @param {string} privateKey
+ * @returns {string} signature
+ */
+const generateTimestampAndSignatureIfNecessary = (data, privateKey) => {
```

Comment:
i see what you're trying to do here by making this route generic enough to have a cache of data -> signatures. but in this case we would only have 1 at any given time right? in that case my take would actually to make this function explicitly for the listen count signatures because I don't think this is a pattern we want to encourage generally

Reply:
can do. why wouldn't we want this generic? what's wrong w/ the pattern? I could see us using this to sign various pieces of information